File: doc-examples/jdbc-example-kotlin/src/main/kotlin/example/Book.kt

package example

import io.micronaut.data.annotation.Relation
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id

@Entity
data class Book(@Id
                @GeneratedValue
                val id: Long,
                val title: String,
                val pages: Int = 0,
                @Relation(value = Relation.Kind.ONE_TO_MANY, mappedBy = "book", cascade = [Relation.Cascade.ALL])
                val reviews: Set<Review>) {
    constructor(id: Long, title: String, pages: Int) : this(id, title, pages, setOf()) {
    }
}

File: doc-examples/jdbc-example-python/src/main/python/example/Book.py
from dataclasses import dataclass, field
from typing import TYPE_CHECKING, Annotated

from micronaut.data.annotation import GeneratedValue, Id, MappedEntity, Relation

if TYPE_CHECKING:
    from example.Review import Review


@MappedEntity
@dataclass
class Book:
    title: str
    pages: int
    reviews: Annotated["list[Review]", Relation(value="ONE_TO_MANY", mappedBy="book", cascade="ALL")] = field(default_factory=list)
    id: Annotated[int | None, Id, GeneratedValue] = None
